### ISSUE TITLE
Use more efficient multiplication of two Taylor models

### ISSUE THREAD

**Issue by dpsanders** [JuliaIntervals/TaylorModels.jl#29]
That does not require duplicating the order of the Taylor series.

**Comment by dpsanders** [comment]
This is in `TaylorN/arithmetic.jl`

**Comment by dpsanders** [comment]
This is usually done by adding a new field `order_bounds` to each Taylor model.
This gives bounds for the homogeneous polynomial at each order.

**Comment by dpsanders** [comment]
How do you feel about that @lbenet ?

**Comment by lbenet** [comment]
I agree that this may avoid duplicating the order, which certainly improves memory allocations and speed. 

I'll try to spend some time on this tonight or tomorrow (I have to give tomorrow a talk); I'll begin trying to upgrade what you had in `TaylorN/arithmetic.jl`.

> This is usually done by adding a new field `order_bounds` to each Taylor model.
This gives bounds for the homogeneous polynomial at each order.

I'm not sure I understand the idea here; do you mean a function to compute the bounds of each `HomogeneousPolynomial`?

**Comment by dpsanders** [comment]
I mean that the bounds are computed when the `TaylorModelN` is defined. Alternatively they can just be computed in the moment, but this will be slower if the object is reused in other calculations.
